src/viewer.ts

```diff
@@ -163,49 +164,76 @@ export class Viewer {
     document.head.appendChild(this.favicon)
   }
 
-  onVimLoaded (vim) {
-    for (let i = 0; i < vim.meshes.length; ++i) {
-      this.meshes.push(vim.meshes[i])
-      this.scene.add(vim.meshes[i])
+  loadInScene (
+    result:
+      | VimScene
+      | THREE.Scene
+      | THREE.Group
+      | THREE.Object3D
+      | THREE.BufferGeometry
+  ) {
+    if (result instanceof VimScene) {
+      this.onVimLoaded(result)
+      return
     }
-    this.boundingSphere = vim.boundingSphere.clone()
+
+    if (result instanceof THREE.Scene) {
+      result.traverse((obj) => {
+        if (obj instanceof THREE.Mesh) this.addToScene(obj)
+      })
+    } else if (result instanceof THREE.BufferGeometry) {
+      result.computeVertexNormals()
+      this.addToScene(new THREE.Mesh(result))
+    } else if (
+      result instanceof THREE.Group ||
+      result instanceof THREE.Object3D
+    ) {
+      this.addToScene(result)
+    }
+    this.boundingSphere = Viewer.computeBoundingSphere(this.scene)
     this.boundingSphere.applyMatrix4(this.getViewMatrix())
+    this.focusModel()
+  }
+
+  onVimLoaded (vim: VimScene) {
     this.vimScene = vim
 
+    for (let i = 0; i < vim.meshes.length; ++i) {
+      this.addToScene(vim.meshes[i])
+    }
+
+    this.boundingSphere = vim.boundingSphere.clone()
+    this.boundingSphere.applyMatrix4(this.getViewMatrix())
     this.focusModel()
   }
 
-  loadFile (fileName, onSuccess: Function) {
-    function getExt (fileName) {
-      const indexOfQueryParams = fileName.lastIndexOf('?')
-      if (indexOfQueryParams >= 0) {
-        fileName = fileName.substring(0, indexOfQueryParams)
-      }
-      const extPos = fileName.lastIndexOf('.')
-      return fileName.slice(extPos + 1).toLowerCase()
-    }
+  addToScene (object: THREE.Object3D) {
+    this.scene.add(object)
+    this.meshes.push(object)
+  }
 
-    console.log('Loading file: ' + fileName)
-    const ext = getExt(fileName)
-    if (ext !== 'vim') {
-      console.error('unhandled file format')
-      return
+  static computeBoundingSphere (scene: THREE.Scene): THREE.Sphere {
+    let sphere = new THREE.Sphere()
+
+    const grow = (geometry: BufferGeometry, matrix: THREE.Matrix4) => {
+      const clone = geometry.clone()
```

Comment:
I would be nervous about this call, does it clone the buffers of the geometry? Also would be much more efficient to apply the transform to the boundingSphere rather than the geometry itself.